Cinema system/Cinema system/Services/AuthenticationService.cs

```diff
@@ -0,0 +1,100 @@
+﻿using CinemaSystem.Interfaces;
+using CinemaSystem.Models;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace CinemaSystem.Services
+{
+    public class AuthenticationService : IAuthenticationService
+    {
+        private readonly CinemaDBContext _db;
+
+        public AuthenticationService(CinemaDBContext db)
+        {
+            _db = db;
+        }
+
+        public bool CheckUserExists(string email)
+        {
+            return _db.Users.Any(u => u.Email == email);
+        }
+
+        public bool CheckCorrectPassword(LoginInfo user)
+        {
+            return user.Password == _db.Users.FirstOrDefault(u => u.Email == user.Email).Password;
+        }
+
+        public bool CheckUserNameUsed(string userName)
+        {
+            return _db.Users.Any(u => u.UserName == userName);
+        }
+
+        public bool CheckCorrectEmail(string email)
+        {
+            return Regex.IsMatch(email,
+                @"^(?("")("".+?(?<!\\)""@)|(([0-9a-z]((\.(?!\.))|[-!#\$%&'\*\+/=\?\^`\{\}\|~\w])*)(?<=[0-9a-z])@))" +
+                @"(?(\[)(\[(\d{1,3}\.){3}\d{1,3}\])|(([0-9a-z][-0-9a-z]*[0-9a-z]*\.)+[a-z0-9][\-a-z0-9]{0,22}[a-z0-9]))$",
+                RegexOptions.IgnoreCase, TimeSpan.FromMilliseconds(250));
+        }
+
+        private string GenerateToken(string userEmail)
+        {
+            //I'm not sure i understand what to use as a key and issuer/audience
+            const string key = "0d5b3235a8b403c3dab9c3f4f65c07fcalskd234n1k41230";
+            var signingKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(key));
+            var credentials = new SigningCredentials(signingKey, SecurityAlgorithms.HmacSha256);
+
+            Claim[] claims = new[] {
+                new Claim(ClaimTypes.Name, userEmail),
+                new Claim(ClaimTypes.Role, _db.Users.FirstOrDefault(u => u.Email == userEmail).Role),
+                new Claim(ClaimTypes.NameIdentifier, _db.Users.FirstOrDefault(u => u.Email == userEmail).Id.ToString())
+            };
+
+            var token = new JwtSecurityToken(
+                issuer:"CinemaSystemServer",
+                audience:"ClientApp",
+                claims: claims,
+                expires: DateTime.Now.AddMinutes(60),
```

Comment:
Why did you choose 60 minutes here?

Reply:
So far, this is just a random token expiration time. As I understand it, it doesn’t matter now and already depends on what kind of project it is.

Reply:
What conditions should you take into consideration when choosing expiration time? Why token should have it at all?

Reply:
Expiration time is used to prevent tokens from being stolen by attackers. 
When choosing at expiration time,  need to pay attention to prevent the theft of the token, but also, if the time is short, the user will often have to enter a username and password.
I also read that this can be resolved with refresh token and sliding expiration for token.
I don’t know exactly if this should be indicated in this project.